### PR TITLE
Cli

### PR DESCRIPTION
**Pull request by breiler**
Added CLI interface for running UGS headless described in #1171

### CODE REVIEW

File: ugs-cli/src/main/java/com/willwinder/ugs/cli/Configuration.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+    /**
+     * Contains the parsed options from the command line arguments
+     */
+    private CommandLine commandLine;
+
+    /**
+     * Default constructor
+     * <p>
+     * Will attempt top create all options based on the values in {@link OptionEnum}.
```
Comment (winder) [inline ugs-cli/src/main/java/com/willwinder/ugs/cli/Configuration.java]: s/top/to

File: ugs-cli/src/main/java/com/willwinder/ugs/cli/TerminalClient.java [hunk 2]
```diff
@@ -0,0 +1,201 @@
+/*
+    Copyright 2016-2018 Will Winder
```
Comment (winder) [inline ugs-cli/src/main/java/com/willwinder/ugs/cli/TerminalClient.java]: s/2016-2018/2019

File: ugs-core/src/com/willwinder/universalgcodesender/listeners/ControllerListener.java [hunk 3]
```diff
@@ -76,9 +76,4 @@ This file is part of Universal Gcode Sender (UGS).
      * Controller status information.
      */
     void statusStringListener(ControllerStatus status);
-    
-    /**
-     * Data gathered while preprocessing commands for queue.
-     */
-    void postProcessData(int numRows);
```
Comment (winder) [inline ugs-core/src/com/willwinder/universalgcodesender/listeners/ControllerListener.java]: Nice catch

File: ugs-core/src/com/willwinder/universalgcodesender/AbstractController.java [hunk 4]
```diff
@@ -868,7 +868,7 @@ public void addListener(ControllerListener listener) {
     @Override
     public void removeListener(ControllerListener listener) {
         if (this.listeners.contains(listener)) {
-            this.listeners.remove(listener);
+            this.listeners.removeIf(l -> l == listener);
```
Comment (breiler) [inline ugs-core/src/com/willwinder/universalgcodesender/AbstractController.java]: Oh, I forgot to mention why I did this. It threw ConcurrentModificationException while shutting down... sometimes. This change got rid of those errors. I'll make sure to add a comment for this.

**Pull request merged**